core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java

```diff
@@ -125,114 +124,120 @@ public void createFromProperties(EventBus eventBus, Properties properties) throw
-                } catch (FrameGrabber.Exception e) {
-                    throw new IllegalStateException("Failed to grab image", e);
-                }
-                final Mat frameMat = convertToMat.convert(videoFrame);
+            final Thread frameExecutor = new Thread(() -> {
+                long lastFrame = System.currentTimeMillis();
+                while (!Thread.interrupted()) {
+                    final Frame videoFrame;
+                    try {
+                        videoFrame = grabber.grab();
+                    } catch (FrameGrabber.Exception e) {
+                        throw new IllegalStateException("Failed to grab image", e);
+                    }
 
-                if (frameMat == null || frameMat.isNull()) {
-                    throw new IllegalStateException("The camera returned a null frame Mat");
-                }
+                    final Mat frameMat = convertToMat.convert(videoFrame);
 
-                frameMat.copyTo(frameOutputSocket.getValue().get());
-                frameOutputSocket.setValue(frameOutputSocket.getValue().get());
-                long thisMoment = System.currentTimeMillis();
-                frameRateOutputSocket.setValue(1000 / (thisMoment - lastFrame));
-                lastFrame = thisMoment;
-            }
-        });
-        frameExecutor.setUncaughtExceptionHandler(
-                (thread, exception) -> {
-                    // TODO Pass Exception to the UI.
-                    System.err.println("Webcam Frame Grabber Thread crashed with uncaught exception:");
-                    exception.printStackTrace();
-                    eventBus.post(new FatalErrorEvent(exception));
-                    try {
-                        stopVideo();
-                    } catch (TimeoutException e) {
-                        System.err.println("Webcam Frame Grabber could not be stopped!");
-                        e.printStackTrace();
-                        eventBus.post(new FatalErrorEvent(e));
+                    if (frameMat == null || frameMat.isNull()) {
+                        throw new IllegalStateException("The camera returned a null frame Mat");
                     }
+
+                    frameMat.copyTo(frameOutputSocket.getValue().get());
+                    frameOutputSocket.setValue(frameOutputSocket.getValue().get());
+                    long thisMoment = System.currentTimeMillis();
+                    frameRateOutputSocket.setValue(1000 / (thisMoment - lastFrame));
+                    lastFrame = thisMoment;
                 }
-        );
-        frameExecutor.setDaemon(true);
-        frameExecutor.start();
-        frameThread = Optional.of(frameExecutor);
+            });
+
+            frameExecutor.setUncaughtExceptionHandler(
+                    (thread, exception) -> {
+                        // TODO Pass Exception to the UI.
+                        System.err.println("Webcam Frame Grabber Thread crashed with uncaught exception:");
```

Comment:
We *do* pass the exception to the UI, right?  If so, don't print the stacktrace here.